bentoml/_internal/runner/container.py

```diff
@@ -544,3 +544,103 @@ def from_batch_payloads(
     ) -> tuple[t.Any, list[int]]:
         container_cls = DataContainerRegistry.find_by_name(payloads[0].container)
         return container_cls.from_batch_payloads(payloads, batch_dim)
+
+
+class TupleContainer(DataContainer[t.Tuple[t.Any], t.Tuple[t.Any]]):
+    @classmethod
+    def batches_to_batch(
+        cls, batches: t.Sequence[t.Tuple[t.Any]], batch_dim: int | t.Tuple = 0
+    ) -> tuple[t.Tuple[t.Any], list[int]]:
+        element_num = len(batches[0])
+        if isinstance(batch_dim, int):
+            batch_dim = (batch_dim,) * element_num
+        else:
+            assert (
+                len(batch_dim) == element_num
+            ), "Tuple DataContainer's batch element number should be equal to batch_dim element number"
+        raw_batch = [
+            AutoContainer.batches_to_batch(element_batches, element_batch_dim)
+            for element_batches, element_batch_dim in zip(zip(*batches), batch_dim)
+        ]
+        batch = tuple(element_batch for element_batch, _ in raw_batch)
+        # following line requires all different container use same indices format
+        indices = raw_batch[0][1]
+        return tuple(batch), indices
```

Comment:
As comment says this part requires all different container using the same indices format. When implementing ndarray, pandas etc. I kept the indices format the same, but need some check for other containers